### DISCUSSION TITLE
Define docker local registry mirror with BuildKit

### DISCUSSION THREAD

**Discussion by Shaked** [docker/buildx#690] [Show and tell]
In order to use a a mirror registry using a local docker registry, you should:

1. Create a [buildkitd.toml file](https://github.com/moby/buildkit/blob/master/docs/buildkitd.toml.md) with the following settings: 
```
$ cat /etc/buildkit/buildkitd.toml
...
    [registry."buildkitd.daemon-ns.svc.cluster.local:5000"]
        http = true
        insecure = true
    [registry."shakedko.azurecr.io"]
        mirrors = ["buildkitd.daemon-ns.svc.cluster.local:5000"]
...
```
2.  Make sure your configuration file is placed at `/etc/buildkit/buildkitd.toml`.
3. Running `docker buildx build -t test .` with the relevant builder will use the local registry to pull the image. 

Related: https://github.com/docker/buildx/issues/516#issuecomment-771236525

**Comment by tonistiigi** [comment]
This isn't quite correct. `/etc/buildkit/buildkitd.toml` needs to be accessible to buildkitd, therefore in buildx it needs to be in buildkit container. Having such file in host where you run buildx doesn't do anything.

Common way  to set config would be to add it in `myconfig.toml` and then set it on create, `docker buildx create --config myconfig.toml`.

**Reply by Shaked** [reply]
> This isn't quite correct. /etc/buildkit/buildkitd.toml needs to be accessible to buildkitd, therefore in buildx it needs to be in buildkit container. Having such file in host where you run buildx doesn't do anything.

I wasn't clear, but I meant that this configuration should be accessible to buildkitd as you mentioned. Otherwise it would not be relevant as you said. 

I have written this doc after making sure it's working :) 

Thank you
Shaked
